src/backend/tests/it/list_users.rs

```diff
@@ -0,0 +1,280 @@
+use std::time::{Duration, UNIX_EPOCH};
+
+use crate::utils::{
+    assertion::assert_user_profiles_eq,
+    mock::{CALLER, ISSUER_CANISTER_ID, VC_HOLDER, VP_JWT},
+    pocketic::{query_call, setup, update_call},
+};
+use candid::Principal;
+use ic_verifiable_credentials::issuer_api::CredentialSpec;
+use pocket_ic::PocketIc;
+use shared::types::user_profile::{
+    AddUserCredentialError, AddUserCredentialRequest, ListUsersRequest, ListUsersResponse,
+    OisyUser, UserProfile,
+};
+
+fn create_users(pic_setup: &(PocketIc, Principal), start: u8, end: u8) -> Vec<OisyUser> {
+    let mut expected_users: Vec<OisyUser> = Vec::new();
+    for i in start..=end {
+        pic_setup.0.advance_time(Duration::new(10, 0));
+        let caller = Principal::self_authenticating(i.to_string());
+        let response = update_call::<UserProfile>(&pic_setup, caller, "create_user_profile", ());
+        let timestamp = pic_setup.0.get_time();
+        let timestamp_nanos = timestamp
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_nanos();
+        let expected_user = OisyUser {
+            updated_timestamp: timestamp_nanos as u64,
```

Comment:
This is fine, but just so that you are aware, in places where wrap-arounds could be critical, it is conventional to do something like:
```
u64::try_from(foo).expect("This conversion should never fail because XYZ")
```

Reply:
But in tests casting is ok, right?